platform/lib/platform/material/media_search.ex

```diff
@@ -36,8 +40,42 @@ defmodule Platform.Material.MediaSearch do
   def changeset(params \\ %{}) do
     data = %{}
 
-    {data, @types}
-    |> Ecto.Changeset.cast(params, Map.keys(@types))
+    new_types =
+      case Map.get(params, "project_id") do
+        nil ->
+          @types
+
+        pid ->
+          case Projects.get_project(pid) do
+            nil ->
+              @types
+
+            project ->
+              Enum.reduce(project.attributes, @types, fn pattr, acc ->
+                attr = ProjectAttribute.to_attribute(pattr)
+                aid = get_attrid(attr)
+
+                case attr.type do
+                  :text ->
+                    acc
+                    |> Map.put(String.to_atom(aid), :string)
+                    |> Map.put(String.to_atom("#{aid}-matchtype"), :string)
+
+                  x when x == :multi_select or x == :select ->
+                    acc |> Map.put(String.to_atom(aid), {:array, :string})
```

Comment:
`String.to_atom/1/` is unsafe (enables DOS); should use `to_existing_atom` instead in all cases.

Reply:
Although I agree that this isn't user-supplied data, it will result in two arbitrary new atoms for each custom attribute. which could result in a DOS.

Perhaps it's possible to use string keys instead?

Reply:
It appears string keys for values in inputs are supported, but only if we choose not to use an `Ecto.Changeset` to store the state of our form: https://hexdocs.pm/phoenix_html/Phoenix.HTML.Form.html#module-access-behaviour

Should I refactor the original changeset to a custom structure that supports string keys?

Reply:
Unfortunately I think that's the best thing to do here; creating two atom for every custom attribute is the sort of thing that, while probably fine today, could turn up as a surprise bottleneck down the road if a lot of projects onboard to the platform. Sorry for the extra work!

Reply:
With `GenericSet`, only existing atoms and strings will be used.